### ISSUE TITLE
Django CKEditor 富文本编辑器

### ISSUE THREAD

**Issue by bjmingbo** [bjmingbo/Learning-Python#172]
https://github.com/django-ckeditor/django-ckeditor#plugins

**Comment by bjmingbo** [comment]
安装后，执行python manage.py collectstatic，报错：

django.core.exceptions.ImproperlyConfigured: You're using the staticfiles app without having set the STATIC_ROOT setting to a filesystem path.

![001](https://user-images.githubusercontent.com/76159285/105991725-22a2cc00-60df-11eb-96c5-60413bf2d9d0.png)

**Comment by bjmingbo** [comment]
设置后，运行命令报错：
Application labels aren't unique, duplicates: staticfiles

![001](https://user-images.githubusercontent.com/76159285/105992504-2d119580-60e0-11eb-8ebb-24d0a0972799.png)

**Comment by bjmingbo** [comment]
参考：https://www.geek-share.com/detail/2704979858.html

![001](https://user-images.githubusercontent.com/76159285/105993839-cbeac180-60e1-11eb-9c10-c593ec8a9f48.png)

![001](https://user-images.githubusercontent.com/76159285/105993884-d9a04700-60e1-11eb-902a-45a2edab1095.png)

还需要导入os，问题解决

![001](https://user-images.githubusercontent.com/76159285/105993961-f2a8f800-60e1-11eb-97c1-5bc2eb775c47.png)

**Comment by bjmingbo** [comment]
![001](https://user-images.githubusercontent.com/76159285/105994934-3e0fd600-60e3-11eb-84d1-41b96e56e71a.png)

**Comment by bjmingbo** [comment]
![001](https://user-images.githubusercontent.com/76159285/105994974-4ff17900-60e3-11eb-83a9-4bd6bf896f34.png)

**Comment by bjmingbo** [comment]
![001](https://user-images.githubusercontent.com/76159285/106070282-21a18700-613f-11eb-9d61-6f6c78c5eb22.png)

![1](https://user-images.githubusercontent.com/76159285/106070287-22d2b400-613f-11eb-9c6c-7f374160d99a.png)

**Comment by bjmingbo** [comment]
![001](https://user-images.githubusercontent.com/76159285/106070407-5ca3ba80-613f-11eb-922b-d926454e880f.png)

![1](https://user-images.githubusercontent.com/76159285/106070411-5dd4e780-613f-11eb-8663-15301fc49de9.png)

**Comment by bjmingbo** [comment]
![001](https://user-images.githubusercontent.com/76159285/106070447-6f1df400-613f-11eb-8e31-9e353f9f64f0.png)

**Comment by bjmingbo** [comment]
![001](https://user-images.githubusercontent.com/76159285/106070904-48ac8880-6140-11eb-9d72-df718b759a62.png)

![1](https://user-images.githubusercontent.com/76159285/106070907-49ddb580-6140-11eb-844b-ad1a6086f633.png)

![003](https://user-images.githubusercontent.com/76159285/106070910-4b0ee280-6140-11eb-9991-366db77fc0d5.png)

**Comment by bjmingbo** [comment]
![001](https://user-images.githubusercontent.com/76159285/106071040-87424300-6140-11eb-8c29-e7aefa0a8856.png)
![1](https://user-images.githubusercontent.com/76159285/106071041-88737000-6140-11eb-9bc9-343950848d39.png)

**Comment by bjmingbo** [comment]
```
CKEDITOR_CONFIGS = {
    'default': {
        'skin': 'moono',
        # 'skin': 'office2013',
        'toolbar_Basic': [
            ['Source', '-', 'Bold', 'Italic']
        ],
        'toolbar_YourCustomToolbarConfig': [
            {'name': 'document', 'items': ['Source', '-', 'Save', 'NewPage', 'Preview', 'Print', '-', 'Templates']},
            {'name': 'clipboard', 'items': ['Cut', 'Copy', 'Paste', 'PasteText', 'PasteFromWord', '-', 'Undo', 'Redo']},
            {'name': 'editing', 'items': ['Find', 'Replace', '-', 'SelectAll']},
            {'name': 'forms',
             'items': ['Form', 'Checkbox', 'Radio', 'TextField', 'Textarea', 'Select', 'Button', 'ImageButton',
                       'HiddenField']},
            '/',
            {'name': 'basicstyles',
             'items': ['Bold', 'Italic', 'Underline', 'Strike', 'Subscript', 'Superscript', '-', 'RemoveFormat']},
            {'name': 'paragraph',
             'items': ['NumberedList', 'BulletedList', '-', 'Outdent', 'Indent', '-', 'Blockquote', 'CreateDiv', '-',
                       'JustifyLeft', 'JustifyCenter', 'JustifyRight', 'JustifyBlock', '-', 'BidiLtr', 'BidiRtl',
                       'Language']},
            {'name': 'links', 'items': ['Link', 'Unlink', 'Anchor']},
            {'name': 'insert',
             'items': ['Image', 'Flash', 'Table', 'HorizontalRule', 'Smiley', 'SpecialChar', 'PageBreak', 'Iframe']},
            '/',
            {'name': 'styles', 'items': ['Styles', 'Format', 'Font', 'FontSize']},
            {'name': 'colors', 'items': ['TextColor', 'BGColor']},
            {'name': 'tools', 'items': ['Maximize', 'ShowBlocks']},
            {'name': 'about', 'items': ['About']},
            '/',  # put this to force next toolbar on new line
            {'name': 'yourcustomtools', 'items': [
                # put the name of your editor.ui.addButton here
                'Preview',
                'Maximize',

            ]},
        ],
        'toolbar': 'YourCustomToolbarConfig',  # put selected toolbar config here
        # 'toolbarGroups': [{ 'name': 'document', 'groups': [ 'mode', 'document', 'doctools' ] }],
        # 'height': 291,
        # 'width': '100%',
        # 'filebrowserWindowHeight': 725,
        # 'filebrowserWindowWidth': 940,
        # 'toolbarCanCollapse': True,
        # 'mathJaxLib': '//cdn.mathjax.org/mathjax/2.2-latest/MathJax.js?config=TeX-AMS_HTML',
        'tabSpaces': 4,
        'extraPlugins': ','.join([
            'uploadimage', # the upload image feature
            # your extra plugins here
            'div',
            'autolink',
            'autoembed',
            'embedsemantic',
            'autogrow',
            # 'devtools',
            'widget',
            'lineutils',
            'clipboard',
            'dialog',
            'dialogui',
            'elementspath'
        ]),
    }
}
```

![001](https://user-images.githubusercontent.com/76159285/106072952-20bf2400-6144-11eb-9435-f863f8546101.png)

![Uploading 1.png…]()
